src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -250,231 +220,94 @@ public void AddPeer(ITxPoolPeer peer)
-                && _transactions.TryGetLast(out var lastTx)
-                && payableGasPrice <= lastTx?.GasBottleneck)
-            {
-                Metrics.PendingTransactionsDiscarded++;
-                Metrics.PendingTransactionsTooLowFee++;
-                if (_logger.IsTrace)
-                    _logger.Trace($"Skipped adding transaction {tx.ToString("  ")}, too low payable gas price.");
-                return AddTxResult.FeeTooLow;
-            }
-
-            if (managedNonce && CheckOwnTransactionAlreadyUsed(tx, currentNonce))
-            {
-                Metrics.PendingTransactionsDiscarded++;
-                if (_logger.IsTrace)
-                    _logger.Trace($"Skipped adding transaction {tx.ToString("  ")}, nonce already used.");
-                return AddTxResult.OwnNonceAlreadyUsed;
-            }
-            
-            return null;
+            return AddCore(tx, startBroadcast);
         }
 
-        private AddTxResult AddCore(Transaction tx, bool isPersistentBroadcast, bool isReorg)
+        private AddTxResult AddCore(Transaction tx, bool isPersistentBroadcast)
         {
-            bool isInHashCache = _hashCache.Get(tx.Hash);
-            // !!! do not change it to |=
-            bool isKnown = !isReorg && isInHashCache;
-
-            if (!isInHashCache)
-            {
-                _hashCache.Set(tx.Hash);
-            }
-
-            /*
-             * we need to make sure that the sender is resolved before adding to the distinct tx pool
-             * as the address is used in the distinct value calculation
-             */
-            if (!isKnown)
+            lock (_locker)
             {
-                lock (_locker)
+                bool eip1559Enabled = _specProvider.GetSpec().IsEip1559Enabled;
+                
+                // TODO: is it called Bottleneck or effective gas price?
+                tx.GasBottleneck = tx.CalculateEffectiveGasPrice(eip1559Enabled, _headInfo.CurrentBaseFee);
+                bool inserted = _transactions.TryInsert(tx.Hash, tx, out Transaction? removed);
+                if (inserted)
                 {
-                    tx.GasBottleneck = tx.CalculateEffectiveGasPrice(_specProvider.GetSpec().IsEip1559Enabled, CurrentBaseFee);
-                    bool inserted = _transactions.TryInsert(tx.Hash, tx, out Transaction? removed);
-                    if (inserted)
-                    {
-                        _transactions.UpdateGroup(tx.SenderAddress, UpdateBucketWithAddedTransaction);
-                        ForgetHashOfRemovedTransaction(removed?.Hash);
-                        Metrics.PendingTransactionsAdded++;
-                        if (tx.IsEip1559)
-                        {
-                            Metrics.Pending1559TransactionsAdded++;
-                        }
-                    }
-                    isKnown |= !inserted;
+                    _transactions.UpdateGroup(tx.SenderAddress!, UpdateBucketWithAddedTransaction);
```

Comment:
Effective gas price is a temporary value. GasBottleneck is calculated for transaction already being in _transactions, it depends from other txs with the same SenderAddress. When adding new transaction to the pool, there is a short period between adding and recalculating bottleneck. We needed some initial value od GasBottleneck. 0 is bad, because transaction can be evicted before being recalculated. UInt256.MaxValue is bad too, because newly added tx will be first-to-include if building block will start. We decided to use here effective gas price - it's overwritten in line 271 (_transaction.UpdateGroup)